nilearn/glm/first_level/first_level.py

```diff
@@ -576,11 +578,27 @@ def fit(
         # Learn the mask
         if self.mask_img is False:
             # We create a dummy mask to preserve functionality of api
-            ref_img = check_niimg(run_imgs[0])
-            self.mask_img = Nifti1Image(
-                np.ones(ref_img.shape[:3]), ref_img.affine
-            )
-        if not isinstance(self.mask_img, NiftiMasker):
+            if isinstance(run_imgs[0], SurfaceImage):
+                left, right = run_imgs[0].mesh.keys()
```

Comment:
there is no reason why the ordering would be left, right ?

Reply:
Yes bad naming. I'll change it. I think we also don't require both hemispheres so maybe a check for that as well